File: quartz-energy/src/backend/QuartzEnergy.Common.Files/Services/IEntityFilesService.cs
namespace QuartzEnergy.Common.Files.Services
{
    using System.Threading.Tasks;

    using QuartzEnergy.Common.Files.Models;

    public interface IEntityFilesService
    {
        Task<UploadedFiles> Upload(UploadFiles uploadFiles);

        Task<BindedToEntity> Bind(BindToEntity bindToEntity);

        Task<DownloadFileResult> Download(DownloadFile downloadFile);
    }
}

File: quartz-energy/src/frontend/src/app/common/services/files/i-entity-files.service.ts
import { BindToEntity } from './../../models/files/bind-to-entity';
import { BindedToEntity } from '../../models/files/binded-to-entity';
import { Task } from '../../tasks/task';
import { InjectionToken } from '@angular/core';

export interface IEntityFilesService {

    bind(bindToEntity: BindToEntity): Task<BindedToEntity>;
}

export const IEntityFilesServiceToken = new InjectionToken<IEntityFilesService>('IEntityFilesService');
